pkg/lib/cockpit-path.ts

```diff
@@ -41,6 +41,8 @@ export function basename(path : string): string {
     const pos = norm.lastIndexOf("/");
     if (pos < 0)
         return norm;
+    else if (pos == 0)
+        return "/";
```

Comment:
This is the same as the basename commandline tool.

For other directory names ending with `/` we have the `drop_slashes()` code above, which indeed special-cases `/`.

Reply:
Exactly, this is the sane option :)